cmake/modules/extensions.cmake

```diff
@@ -493,6 +493,92 @@ function(zephyr_library_compile_definitions item)
   target_compile_definitions(${ZEPHYR_CURRENT_LIBRARY} PRIVATE ${item} ${ARGN})
 endfunction()
 
+#
+# Helper functions for specifying _POSIX_C_SOURCE and _XOPEN_SOURCE application conformance.
+#
+
+function(target_posix_library_compile_options target)
+  set(min_level 0)
+
+  if(CONFIG_POSIX_C_SOURCE_CHOICE_NONE)
+    # Do not specify any application conformance level
+  endif()
+
+  # Check if a minimum application conformance level is required
+  if(ARGC GREATER_EQUAL 2)
+    # strip out L if it exists in the argument
+    string(REPLACE "L" "" min_level ${ARGV1})
+    # ensure that ARGV0 can be evaluated as a number
+    math(EXPR min_level "${min_level} - 0" OUTPUT_FORMAT DECIMAL)
+  endif()
+
+  if(CONFIG_POSIX_C_SOURCE_CHOICE_DEFAULT)
+    set(level CONFIG_POSIX_C_SOURCE_LATEST_VERSION)
+  elseif(CONFIG_POSIX_C_SOURCE_CHOICE_200809L)
+    set(level 200809)
+  elseif(CONFIG_POSIX_C_SOURCE_CHOICE_200112L)
+    set(level 200112)
+  elseif(CONFIG_POSIX_C_SOURCE_CHOICE_199506L)
+    set(level 199506)
+  elseif(CONFIG_POSIX_C_SOURCE_CHOICE_199309L)
+    set(level 199309)
+  elseif(CONFIG_POSIX_C_SOURCE_CHOICE_2)
+    set(level 2)
+  elseif(CONFIG_POSIX_C_SOURCE_CHOICE_1)
+    set(level 1)
+  endif()
+
+  if(level LESS min_level)
+    set(level min_level)
+  endif()
+
+  if(level GREATER 2)
+    # append L (required for _POSIX_C_SOURCE > 2)
+    set(level "${level}L")
+  endif()
+
+  target_compile_options(${target} PRIVATE -U_POSIX_C_SOURCE -D_POSIX_C_SOURCE=${level})
```

Comment:
this would define an empty `POSIX_C_SOURCE` if `${level}` is none (`CONFIG_POSIX_C_SOURCE_CHOICE_NONE`)

Reply:
Ah - i forgot to `return()` when checking for none (that was in a previous revision).